src/thor/matrix_action.cc

```diff
@@ -42,10 +58,10 @@ std::string thor_worker_t::matrix(Api& request) {
                                       mode, max_matrix_distance.find(costing)->second);
   };
   auto timedistancematrix = [&]() {
-    return time_distance_matrix_.SourceToTarget(options.sources(), options.targets(), *reader,
-                                                mode_costing, mode,
-                                                max_matrix_distance.find(costing)->second,
-                                                options.matrix_locations());
+    return time_distance_matrix_.SourceToTarget(*options.mutable_sources(),
+                                                *options.mutable_targets(), *reader, mode_costing,
+                                                mode, max_matrix_distance.find(costing)->second,
```

Comment:
we need mutable locations here for the TimeInfo object to be created properly with `current` time